### PR TITLE
1907 changes

### PR DESCRIPTION
**Pull request by soumyapattnaik**

### CODE REVIEW

File: VMBackup/main/Utils/HandlerUtil.py [hunk 1]
```diff
@@ -684,6 +687,13 @@ def get_shell_script_log(self):
             self.log("Can't receive shell log file: " + str(e))
             return lines
 
+    def update_settings_file(self):
+        self.log("removing the protected settings" + json.dumps(self._context._config))
```
Comment (soumyapattnaik) [inline VMBackup/main/Utils/HandlerUtil.py]: the config value log is intentionally kept. i will remove it before final checkin, after all PR comments are resolved

File: VMBackup/main/handle.py [hunk 2]
```diff
@@ -480,13 +479,15 @@ def daemon():
         hutil.SetExtErrorCode(run_result) #setting extension errorcode at the end if missed somewhere
         HandlerUtil.HandlerUtility.add_to_telemetery_data("extErrorCode", str(ExtensionErrorCodeHelper.ExtensionErrorCodeHelper.ExtensionErrorCodeNameDict[hutil.ExtErrorCode]))
         total_used_size = -1
+        hutil.update_settings_file()
```
Comment (vityagi) [inline VMBackup/main/handle.py]: This should be at line number 458.
Not needed here.

File: VMBackup/main/handle.py [hunk 3]
```diff
@@ -480,13 +479,15 @@ def daemon():
...
         blob_report_msg, file_report_msg = get_status_to_report(run_status,run_result,error_msg, snapshot_info_array)
         if(hutil.is_status_file_exists()):
             status_report_to_file(file_report_msg)
         status_report_to_blob(blob_report_msg)
     except Exception as e:
         errMsg = 'Failed to log status in extension'
         backup_logger.log(errMsg, True, 'Error')
+        hutil.update_settings_file()
```
Comment (vityagi) [inline VMBackup/main/handle.py]: Not required here

File: VMBackup/main/handle.py [hunk 4]
```diff
@@ -153,8 +153,9 @@ def get_status_to_report(status, status_code, message, snapshot_info = None):
     return blob_report_msg, file_report_msg
 
 def exit_with_commit_log(status,result,error_msg, para_parser):
-    global backup_logger
+    global backup_logger, hutil
     backup_logger.log(error_msg, True, 'Error')
+    hutil.update_settings_file()
```
Comment (vityagi) [inline VMBackup/main/handle.py]: Not needed here

File: VMBackup/main/guestsnapshotter.py [hunk 5]
```diff
@@ -252,6 +264,8 @@ def snapshotall_parallel(self, paras, freezer, thaw_done, g_fsfreeze_on):
         except Exception as e:
             errorMsg = " Unable to perform parallel snapshot with error: %s, stack trace: %s" % (str(e), traceback.format_exc())
             self.logger.log(errorMsg)
+            self.logger.log("Setting to sequential snapshot")
+            self.hutil.set_value_to_configfile('seqsnapshot', '1')
```
Comment (vityagi) [inline VMBackup/main/guestsnapshotter.py]: exception can be due to many reasons, we should not set to Seq here. We should have try-catch around the exact code where we know it is due to mp-process.

**Comment by vityagi** [comment]
Looks good to merge

**Pull request merged**